src-tauri/capabilities/migrated.json

```
@@ -0,0 +1,74 @@
+{
+  "identifier": "migrated",
+  "description": "permissions that were migrated from v1",
+  "local": true,
+  "windows": [
+    "main"
+  ],
+  "permissions": [
+    "core:default",
+    "fs:allow-read-file",
+    "fs:allow-write-file",
+    "fs:allow-read-dir",
+    "fs:allow-copy-file",
+    "fs:allow-mkdir",
+    "fs:allow-remove",
+    "fs:allow-remove",
+    "fs:allow-rename",
+    "fs:allow-exists",
+    {
+      "identifier": "fs:scope",
+      "allow": [
+        "$APPDATA/**"
+      ]
+    },
```

Does this work on Linux?

---

I think so, it's able to read and write to my config, i'll check again tho

---

can confirm, $APPDATA points to [appDataDir](https://v2.tauri.app/reference/javascript/api/namespacepath/#appdatadir) from '@tauri-apps/api/path'
it results in /home/pie/.local/share/com.cursescaptions.app
i'm gonna mark this as resolved and if we have any file system issues later down the line we'll add other paths to it

---

Still unsure if this is enough.
On linux it saves config in other places as well iirc.
Considering we didn't have a fs scope before i think we can blanket allow everything for now and add some scoping back later if we consider it necessity.

---

Will do

---

I've done some digging.
It seems we cannot just ALLOW_ALL or smth of the sort same as with the tauri command permissions

But this should suffice(maybe be even overkill but that way we dont have to think about it much) as we only use BaseDirectory.AppData as parent dir when accessing files from frontend and backend is exempt from the filter anyways.
```suggestion
    "fs:scope-app-recursive",
```